sdk/typescript/src/verify/index.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { PublicKey, SerializedSignature } from '../cryptography/index.js';
+import { parseSerializedSignature } from '../cryptography/index.js';
+import { Ed25519PublicKey } from '../keypairs/ed25519/publickey.js';
+import { Secp256k1PublicKey } from '../keypairs/secp256k1/publickey.js';
+import { Secp256r1PublicKey } from '../keypairs/secp256r1/publickey.js';
+
+export async function verifySignature(
+	bytes: Uint8Array,
+	signature: SerializedSignature,
+): Promise<false | PublicKey> {
```

Comment:
Not sure if this is a good way to handle returns of these methods, but I think getting back the public key is useful here so you can also validate that the key used to sign a message was the expected one

Reply:
i dont have strong preference but slight preference just having it to return true/false is more intuitive and easier to return to the outer level

Reply:
Rather than false | publicKey I wonder if this should just throw? I do agree getting the public key back is pretty useful, because often you want to validate ownership of an address and without the return here it becomes multi-step.

Reply:
updated this to throw